include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -293,11 +293,52 @@ struct __parallel_scan_submitter;
 template <typename _CustomName, typename... _PropagateScanName>
 struct __parallel_scan_submitter<_CustomName, __internal::__optional_kernel_name<_PropagateScanName...>>
 {
+    static constexpr int
+    __iters_per_item(::oneapi::dpl::__internal::__algorithm_type __algo_type)
+    {
+        using namespace ::oneapi::dpl::__internal;
+        switch (__algo_type)
+        {
+        case __algorithm_type::partition:
+            return 1;
+        case __algorithm_type::unique:
+            return 2;
+        case __algorithm_type::set_operation:
+            return 4;
+        case __algorithm_type::scan_copy:
+            return 16;
+        case __algorithm_type::transform_scan:
+            return 32;
+        default:
```

Comment:
Better to remove ```default``` branch and call ```return 16;``` after switch/case: we will have compile warnings or errors if we extend this enum.